src/main/java/filter/expression/Qualifier.java

```diff
@@ -93,6 +93,44 @@ public Qualifier(String name, List<SortKey> keys) {
         this.sortKeys = new ArrayList<>(keys);
     }
 
+    public static FilterExpression replaceMilestoneAliases(IModel model, FilterExpression expr) {
+        List<String> repoIds = getContentOfMetaQualifier(expr, REPO).stream()
+                .map(repoId -> repoId.toLowerCase())
+                .collect(Collectors.toList());
+
+        if (repoIds.isEmpty()) {
+            repoIds.add(model.getDefaultRepo().toLowerCase());
+        }
+
+        List<TurboMilestone> allMilestones = model.getMilestones().stream()
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .sorted(Qualifier::milestoneDueDateComparator)
+                .collect(Collectors.toList());
+
+        Optional<Integer> currentMilestoneIndex = getCurrentMilestoneIndex(allMilestones);
+
+        if (!currentMilestoneIndex.isPresent()) {
+            return expr;
+        }
```

Comment:
Here is the check. And I changed the parameter of `convertMilestoneAliasQualifier` from `Optional<Integer>` to `int`.